src/test/SerializeTool_test.cpp

```diff
@@ -0,0 +1,573 @@
+            boost::optional<std::string> const& seed)
+        {
+            std::stringstream capture;
+            CoutRedirect coutRedirect{ capture };
+
+            doCreateKeyfile(keyFile, kt, seed);
+
+            auto const key = RippleKey::make_RippleKey(keyFile);
+
+            auto const known = boost::str(
+                boost::format("New ripple key created in %s.\n"
+                "Key type is %s, and account ID is %s.\n"
+                "\nThis file should be stored securely and not shared.\n\n")
+                % keyFile.string()
+                % to_string(key.keyType())
+                % toBase58(calcAccountID(key.publicKey())));
+
+            // Test that the function will not overwrite
+            try
+            {
+                doCreateKeyfile(keyFile, {}, {});
+                fail();
+            }
+            catch (std::exception const& e)
+            {
+                BEAST_EXPECT(e.what() ==
+                    std::string{ "Refusing to overwrite existing key file: " +
+                    keyFile.string()});
+            }
+
+            remove(keyFile);
+
+            BEAST_EXPECT(capture.str() == known);
+        };
+
+        test(boost::none, boost::none);
+        test(boost::none, std::string{ "masterpassphrase" });
+        test(std::string(to_string(KeyType::ed25519)), boost::none);
+        test(std::string(to_string(KeyType::secp256k1)),
+            std::string{ "alice" });
+
+        // edge cases
+        {
+            // invalid keytype
+            std::stringstream capture;
+            CoutRedirect coutRedirect{ capture };
+
+            doCreateKeyfile(keyFile, std::string{ "NSA special" },
+                boost::none);
+
+            BEAST_EXPECT(!exists(keyFile));
+            BEAST_EXPECT(capture.str() == "Invalid key type: \"NSA special\"\n");
+        }
+        {
+            // empty seed
+            try
+            {
+                doCreateKeyfile(keyFile, std::string{ "ed25519" },
+                    std::string{ "" });
+                fail();
+            }
+            catch (std::exception const& e)
+            {
+                BEAST_EXPECT(e.what() ==
+                    std::string{ "Unable to parse seed: " });
+            }
+
+            BEAST_EXPECT(!exists(keyFile));
+        }
+    }
+
+    void
+    testRunCommand ()
+    {
+        testcase ("Run Command");
+
+        std::stringstream capture;
+        CoutRedirect coutRedirect{ capture };
+
+        using namespace boost::filesystem;
+
+        std::string const subdir = "test_key_file";
+        KeyFileGuard g(*this, subdir);
+        path const keyFile = subdir / ".ripple" / "secret-key.txt";
+
+        auto testCommand = [&](
+            std::string const& command,
+            std::vector <std::string> const& args,
+            std::string const& expectedError)
+        {
+            try
+            {
+                runCommand(command, args, keyFile, {}, {});
```

Comment:
It looks like there isn't a test for passing in the key type and/or seed to `runCommand`